### PR TITLE
store: refactor and improve store cache

### PR DESCRIPTION
**Pull request by bboozzoo**
Refactor the store cache manager cleanup to make it easier to follow and do not keep repeating the same operations multiple times.

Thanks for helping us make a better snapd!
Have you signed the [license agreement](https://www.ubuntu.com/legal/contributors) and read the [contribution guide](https://github.com/snapcore/snapd/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

**Review by MiguelPires — APPROVED** [review]
lgtm, thanks

File: store/cache.go [hunk 1]
```diff
@@ -165,54 +165,43 @@ func (cm *CacheManager) cleanup() error {
 		return err
 	}
 
-	// we need the modtime so convert to FileInfo
-	fil := make([]os.FileInfo, 0, len(entries))
+	if len(entries) <= cm.maxItems {
```
Comment (zyga) [inline store/cache.go:168]: Smart!

**Review by zyga — APPROVED** [review]
The change looks good. 

Do we need a new test for this? I was expecting something not to be covered (hence the issue). Perhaps it's worth adding something?

File: store/cache.go [hunk 2]
```diff
@@ -165,54 +165,43 @@ func (cm *CacheManager) cleanup() error {
...
+		return nil
+	}
+
+	var pruneCandidates []os.FileInfo
```
[the remark this answers is not in the document]
Comment (bboozzoo) [reply store/cache.go:172]: Thanks for the comment. It was intentional, there should be few entries meeting the criteria. However, I'm thinking maybe it makes sense to preallocate the slice for say 1/4 or 1/5 length of entries.

[the remark this answers is not in the document]
Comment (bboozzoo) [reply store/cache.go]: Pushed a tweak for this. Please have a look

**Comment by bboozzoo** [comment]
> The change looks good.
> 
> Do we need a new test for this? I was expecting something not to be covered (hence the issue). Perhaps it's worth adding something?

There are tests covering this already. The branch is simply a small improvement I did while investigating another problem around the cache and pre-downloads.

**Comment by bboozzoo** [comment]
Failing tests:
- google:ubuntu-20.04-64:tests/main/nvidia-files:550_server - known issue with LP

**Pull request merged**